util/src/main/java/tech/pegasys/teku/util/async/DelayedExecutorAsyncRunner.java

```diff
@@ -58,15 +56,23 @@ public static DelayedExecutorAsyncRunner create(Executor executor) {
   <U> SafeFuture<U> runAsync(final Supplier<SafeFuture<U>> action, final Executor executor) {
     final SafeFuture<U> result = new SafeFuture<>();
     try {
-      executor.execute(() -> propagateResult(action.get(), result));
```

Comment:
What I understand is that `propagateResult`, which dealt with exceptions that return failed future correctly, did not deal well with exceptions that do not return a failed future. Thus you've implemented runTask to account for any throwable that might occur. Did I get that correctly?

Reply:
Yep.